akka-http/src/main/scala/akka/http/javadsl/server/HttpApp.scala

```diff
@@ -16,7 +16,7 @@ import akka.http.scaladsl.Http.ServerBinding
 abstract class HttpApp
   extends AllDirectives
   with HttpServiceBase {
-  protected def createRoute(): Route
+  protected[javadsl] def createRoute(): Route
```

Comment:
sure? In java people will be able to extend it because it compiles down to `public` I believe, but what was the intent here?